src/helpers/contexts/SettingsContext.tsx

```diff
@@ -13,13 +14,19 @@ interface Settings {
         snapToGridAmount: number,
         setSnapToGridAmount: React.Dispatch<React.SetStateAction<number>>
     ];
+
+    // Node Settings
+    useNodeFavorites: readonly [string[], React.Dispatch<React.SetStateAction<string[]>>];
```

Comment:
Could you make the type `readonly string[]`? This will make sure nobody can modify the shared state.
```suggestion
    useNodeFavorites: readonly [readonly string[], React.Dispatch<React.SetStateAction<readonly string[]>>];
```